xbmc/utils/FileExtensionProvider.cpp

```diff
@@ -111,16 +132,30 @@ void CFileExtensionProvider::SetAddonExtensions()
 void CFileExtensionProvider::SetAddonExtensions(const TYPE& type)
 {
   std::vector<std::string> extensions;
+  std::vector<std::string> fileFolderExtensions;
   BinaryAddonBaseList addonInfos;
   CServiceBroker::GetBinaryAddonManager().GetAddonInfos(addonInfos, true, type);
   for (const auto& addonInfo : addonInfos)
   {
-    std::string ext = addonInfo->Type(type)->GetValue(ADDON_VFS == type ? "@extensions" : "@extension").asString();
+    std::string info = ADDON_VFS == type ? "@extensions" : "@extension";
+    std::string ext = addonInfo->Type(type)->GetValue(info).asString();
     if (!ext.empty())
       extensions.push_back(ext);
+    if (type == ADDON_VFS || type == ADDON_AUDIODECODER)
```

Comment:
Enclose the code below into `if (!ext.empty())`, we can skip it if ext is empty.